lighthouse-core/audits/byte-efficiency/unused-javascript.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./byte-efficiency-audit');
+const URL = require('../../lib/url-shim');
+
+const IGNORE_THRESHOLD_IN_BYTES = 2048;
+
+class UnusedJavaScript extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'JavaScript',
+      name: 'unused-javascript',
+      description: 'Unused JavaScript',
+      informative: true,
+      helpText: 'Try to elinimate unused script. ',
+      requiredArtifacts: ['JsUsage', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {{used: number, unused: number}} objA
+   * @param {{used: number, unused: number}} objB
+   * @return {{used: number, unused: number}}
+   */
+  static _add(objA, objB) {
+    return {
+      used: objA.used + objB.used,
+      unused: objA.unused + objB.unused,
+    };
+  }
+
+  /**
+   * Determines the length of unused and used portions in a set of ranges.
+   * Because the protocol reports usage for each function and ranges within each function, nested
+   * regions are represented numerous times. This function makes sure to only count each string
+   * section once. See unit and smoke tests for examples.
+   *
+   * @param {!Array<{isUsed: boolean, startOffset: number, endOffset: number
+   *    used: number, unused: number}>} ranges
+   * @return {{used: number, unused: number}}
+   */
+  static findUsedAndUnused(ranges) {
```

Comment:
yeah this was a major pain, and once we get up-to-date I'd prefer to have a canonical place to consume this rather than maintain in Lighthouse too, but for now I think it's worth getting going